### PR TITLE
chore: onboard a utility for getting the test data for manual testing

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

Added utility to get test data for manual testing

## What is the related Linear task?

Resolves INT-1567

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new option in integration tests to specify the mode of operation, enhancing test data handling.
- **Chores**
	- Updated `.gitignore` to exclude temporary files, improving file management.

### CODE REVIEW

**Comment by utsabc** [comment]
@coderabbitai revie

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3092/test-report.html

File: test/integrations/component.test.ts [hunk 1]
```diff
@@ -222,23 +224,60 @@ describe.each(allTestDataFilePaths)('%s Tests', (testDataPath) => {
...
+      });
     });
-  });
+
+    it('data generated successfully', () => {
+      // write the data to a file
+      
+      // create directory if not exists
+      const dir = join(__dirname, '../../temp');
+      if (!require('fs').existsSync(dir)) {
```
[the remark this answers is not in the document]
Comment (utsabc) [reply test/integrations/component.test.ts]: addressed

File: test/integrations/component.test.ts [hunk 2]
```diff
@@ -222,23 +224,60 @@ describe.each(allTestDataFilePaths)('%s Tests', (testDataPath) => {
...
-  describe(`${testData[0].name} ${testData[0].module}`, () => {
-    test.each(testData)('$feature -> $description', async (tcData) => {
-      tcData?.mockFns?.(mockAdapter);
 
-      switch (tcData.module) {
-        case tags.MODULES.DESTINATION:
-          await destinationTestHandler(tcData);
+  if (mode === 'passive') {
+    // print the test data
+    const result: any = [];
```
[the remark this answers is not in the document]
Comment (utsabc) [reply test/integrations/component.test.ts]: addressed

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3092/test-report.html

**Review by koladilip — APPROVED** [review]

File: test/integrations/component.test.ts [hunk 3]
```diff
@@ -222,23 +224,60 @@
...
     });
-  });
+
+    it('data generated successfully', () => {
+      // write the data to a file
+      
+      // create directory if not exists
+      const dir = join(__dirname, '../../temp');
+      if (!require('fs').existsSync(dir)) {
+        require('fs').mkdirSync(dir);
```
[the remark this answers is not in the document]
Comment (koladilip) [reply test/integrations/component.test.ts]: I was talking about this in my earlier comment

File: test/integrations/component.test.ts [hunk 4]
```diff
@@ -222,23 +224,60 @@
...
+      }
+      appendFileSync(join(__dirname, '../../temp/test_data.json'), JSON.stringify(result, null, 2));
+      console.log('Data generated successfully at temp/test_data.json');
+      expect(true).toEqual(true); // Intentionally pass the test case
+    });
```
[the remark this answers is not in the document]
Comment (koladilip) [reply test/integrations/component.test.ts]: should we write some other util for generation based on this comment? @utsabc

File: test/integrations/component.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (koladilip) [reply test/integrations/component.test.ts]: basically, I was also saying that @utsabc should put existsSync and mkdirSync in top imports

**Pull request closed without merging**